apps/lead/filter_set.py

```diff
@@ -527,6 +532,27 @@ def _entry_subquery(entry_qs: models.QuerySet):
         # Call super function
         return super().filter_queryset(qs)
 
+    def duplicates_of_filter(self, qs, _, lead_id: int):
+        if lead_id is None:
+            return qs
+        return qs.filter(
+            models.Q(duplicate_leads=lead_id) |
+            models.Q(duplicate_of=lead_id)
+        )
+
+    def has_duplicates_filter(self, qs, _, val: bool):
+        if val is True:
+            return qs.filter(
+                models.Q(duplicate_leads__isnull=False) |
+                models.Q(duplicate_of__isnull=False),
+            )
+        elif val is False:
+            return qs.filter(
+                duplicate_leads=None,
+                duplicate_of=None,
+            )
+        return qs
```

Comment:
We can use the annotated count here?

Reply:
That's a good point. Will test and push.